### ISSUE TITLE
Admin\Facilities - External User can add a facility

### ISSUE THREAD

**Issue by zoeyli-46** [bcgov/cas-registration#1361]
#### Description:
As an External User, I will fill out the registration information required for my facilities

Blocked by: #1331 

#### Acceptance Criteria:

Given that I am on the facilities table
When I click add facility
I am brought to the form for adding a new facility
[Form](https://www.figma.com/file/LsDmLDCdnJqI8UahyqvOD8/OBPS-App-Designs?type=design&node-id=8638-2373&mode=design&t=oyOuuNED6BMrtECN-4) 
- [ ] SFO facility form [Figma](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/OBPS-App-Designs?node-id=8471-1763&t=M2WIeqIxVnQwNDxr-4)
- [ ] LFO facility form [Figma](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/OBPS-App-Designs?node-id=9731-1223&t=M2WIeqIxVnQwNDxr-4)

#### Development Checklist:

- [ ] 

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:
The single step form component expects nested schemas. We looked nested vs. unnested schemas with some directing questions:
- What do nested schemas on the front end provide? Are they enabling a requirement to be met that couldn't be met by flat schemas?
Nested schemas provide section-by-section live validation. There may be a way to do this with flat schemas but we don't have a POC at the moment. We do have a way of visually sectioning forms with flat schemas we used in CIF via object templates.
- If both can solve the problem, how do the implementation and long term maintenance costs compare? Track a hypothetical structural change in the database through both approaches and see what kind of effort each requires. Similarly, track a change to the presentation and what that would involve
If a new field is added or removed, we have to update the db table and ninja schemas no matter which schema approach we use. 
If we're using nested schemas, we need to update the jsonSchema and possibly uiSchema (if the field had ui needs).
With an object template, we update jsonSchema, possibly uiSchema, and the list of which field goes in what groups.
- What limitations will the generic schema flattener impose? Don’t confine this question to the frontend, follow it the whole way down. Over generalization has cost us heavily on other projects, including dealing with nested schemas.
Since everything going in and out of the BE is flat, the generic flattener/unflattener should only affect the FE. Limitations: the jsonSchema has to be sectioned and the formData has to be flat for the function to work so no subsections. The single step component is heavily typed to the sectioned jsonSchema (display, validation, and formData processing is all based on the sectioned schema), and therefore the FE tests are too, but the rest of the app isn't constrained. Giraffe only has multi-step forms, not single step, so it shouldn't affect them.

**Comment by shon-button** [comment]
@patriciarussellCAS

Reachable at:
https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/administration/operations/002d5a9e-32a6-4191-938c-2c02bfec592d/facilities/add-facility?operations_title=Operation%202

**Comment by shon-button** [comment]
@patriciarussellCAS 

Reachable at:
https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/administration/operations/002d5a9e-32a6-4191-938c-2c02bfec592d/facilities/add-facility?operations_title=Operation%202

**Comment by patriciarussellCAS** [comment]
Looks good. 
My notes for design backlog: 
- Should a back button be added? 
- All facilities need to be located in BC - I don't think we should have a drop-down with other options. They should only be able to choose BC.

**Comment by zoeyli-46** [comment]
Hi @patriciarussellCAS! Will a back button be necessary if its only one page? If they click "cancel" then they will be brought back to the grid, so it functions the same. Or, we could replace cancel with "back"?

**Comment by patriciarussellCAS** [comment]
If you've added a facility and hit submit, what would you do next? Do we expect users to know that the only way out of this page now is to use the breadcrumbs on top? 
It looks like this is common throughout the app so must be how we're doing it. In future we may want to consider including a "back to facilities table" button or something that guides them back to where they would logically go next.
